### DISCUSSION TITLE
Abnormal segmentation prediction without pred_mask

### DISCUSSION THREAD

**Discussion by pirnerjonas** [open-edge-platform/anomalib#2263] [Q&A]
Hi,

I just encountered a situation where I predict abnormal for an image but pred_mask doesn't show anything (`image_result.pred_mask.sum()` is 0):

![image](https://github.com/user-attachments/assets/695aa8a1-3f8e-43a0-bd20-5ff098d0b99c)

is this to be expected?

**Comment by yxteo2** [comment]
Can show how you get the predicted mask?

**Comment by alexriedel1** [comment]
Which type of model are you using?

what you see probably happens because the threshold for an whole Image to be anomalous and the treshold for a pixel on the anomaly map to be anomalous are calculated separately. 
You can have a deeper look into these thresholds to find out if that is the case

**Reply by pirnerjonas** [reply]
I think for that test it was Padim. And yes I think it was like you said @alexriedel1. Image threshold influences pred label, pixel threshold influences pred mask.
